corehq/blobs/mixin.py

```diff
@@ -57,32 +76,55 @@ class Meta(object):
 
     _atomic_blobs = None
 
-    def _blobdb_bucket(self):
-        if self._id is None:
-            raise ResourceNotFound(
-                "cannot manipulate attachment on unidentified document")
-        return join(_get_couchdb_name(type(self)), safe_id(self._id))
+    @classmethod
+    def wrap(cls, data):
+        if data.get("external_blobs"):
+            doc_id = safe_id(data["_id"])
+            dbname = _get_couchdb_name(cls)
+            normalize = BlobMetaPointer._normalize_json
+            blobs = {}
+            normalized = False
+            for key, value in data["external_blobs"].items():
+                if value["doc_type"] == "BlobMetaPointer":
+                    blobs[key] = value
+                else:
+                    blobs[key] = normalize(dbname, doc_id, value)
+                    normalized = True
+            if normalized:
+                data = data.copy()
+                data["external_blobs"] = blobs
+        return super(BlobMixin, cls).wrap(data)
```

Comment:
Looks like we are to consider just the ones that don't have `doc_type` as `BlobMetaPointer` and normalize them
can we re-write this as the following or a better version
```
if data.get("external_blobs"):
    if any(value["doc_type"] != "BlobMetaPointer" for key,value in data.get("external_blobs").items()):
        data = data.copy()
        doc_id = safe_id(data["_id"])
        dbname = _get_couchdb_name(cls)
        for key, value in data["external_blobs"].items():
            if value["doc_type"] != "BlobMetaPointer":
                data["external_blobs"][key] = normalize(dbname, doc_id, value)
return super(BlobMixin, cls).wrap(data)
```
to avoid iterating each time.

Also is this like a migration process? keep updating as an when accessed?